File: python/src/shapes/base/commons/base_shape.py
from abc import ABC, abstractmethod


class BaseShape(ABC):
    def __init__(self, dimensions: list[dict[str, float]]):
        """
        Initialize the BaseShape with a list of dictionaries representing dimensions.
        Each dictionary should have a single key-value pair, e.g., `[{"width":
        12}, {"length": 15}]`.
        """
        self._dimensions = dimensions

    @property
    def dimensions(self) -> list[dict[str, float]]:
        return self._dimensions

    def show_available_dimension_count(self) -> int:
        return len(self.dimensions)

    @staticmethod
    def validate_shape() -> bool:
        return True

    @abstractmethod
    def print_description(self) -> None: ...

File: java/src/main/java/shapes/base/commons/BaseShape.java
package shapes.base.commons;

import java.util.List;
import java.util.Map;

public abstract class BaseShape {
    private final List<Map<String, Double>> dimensions;

    /**
     * Initialize the BaseShape with a list of dictionaries representing dimensions.
     * Each dictionary should have a single key-value pair, e.g.,
     * <code>[{"width": 12}, {"length": 15}]</code>.
     */
    public BaseShape(List<Map<String, Double>> dimensions) {
        this.dimensions = dimensions;
    }

    public List<Map<String, Double>> getDimensions() {
        return dimensions;
    }

    public int getDimensionAvailable() {
        return dimensions.size();
    }

    public static boolean isAShape() {
        return true;
    }

    public abstract void printDescription();
}
